manim/mobject/svg/tex_mobject.py

```diff
@@ -47,13 +47,18 @@ class SingleStringMathTex(SVGMobject):
         "organize_left_to_right": False,
         "alignment": "",
         "type": "tex",
+        "template": config["tex_template"],
```

Comment:
Issue: please do not use `config` to define a key/value of `CONFIG`. Using the global config dict at the time when manim is imported (i.e. at the time all these classes are defined) causes all sorts of problems, and it is one of the main barriers to getting rid of CONFIG dicts.

Can you set `template` to `None` here and then use `config[tex_template]` in `__init__`?

Reply:
Add `"template": None`
and then in `__init__`  do something like `if self.template is None: self.template = config["tex_template"]`?

Reply:
Yes, or whatever is logically equivalent to setting the template in CONFIG. I think you would have to do `if self.template is None: self.template = kwargs.get("tex_template", config["tex_template"]).

Reply:
I am getting confused between self.x, keyword argument x, and config["x"], when are these the same, when do they get set, etc.

Reply:
I feel you. This is why CONFIG must go away :)

CONFIG is used to set default values for many attributes, but it can be overriden by passing kwargs. So, usually in the codebase, config[opt] is the default value of CONFIG, which is then used only when there is no corresponding kwarg (or when a inherited class overrides it). But we're getting rid of all of that, so let's just do the usual thing of having an attribute self.x with a default value that can be overriden through kwargs.

Reply:
https://github.com/ManimCommunity/manim/pull/457#discussion_r492743496

and is the before or after `digest_configs()`?

Reply:
done